### PR TITLE
Feature/copy paste

### PR DESCRIPTION
**Pull request by morloy**
Adds basic support for copy & paste to and from existing spreadsheet applications as requested in #51.

### CODE REVIEW

**Comment by iddan** [comment]
Hey, seems like there were multiple changes in this PR. Can you describe what are the changes made?

File: src/actions.js [hunk 1]
```diff
@@ -127,11 +127,15 @@ export const cut = (state: Types.StoreState<*>) => ({
...
-    return null;
-  }
-  const minPoint = PointSet.min(state.copied);
+export async function paste<Cell: Types.CellBase>(
+  state: Types.StoreState<Cell>
+) {
+  const text = await readTextFromClipboard();
+  if (!text) return null;
+  const matrix = Matrix.parse(text);
+  const copied = PointMap.fromMatrix<any>(matrix);
```
Comment (iddan) [inline src/actions.js]: No need to provide `any` here.
If Flow insists on a type argument it should be `Cell`.

File: src/Spreadsheet.js [hunk 2]
```diff
@@ -115,12 +114,6 @@ class Spreadsheet<CellType, Value> extends PureComponent<{|
 
   formulaParser = new FormulaParser();
 
-  /**
-   * Internally used value to check if the copied text match the live objects
-   * inside state.copied
-   */
-  _clippedText: string | null = null;
```
Comment (iddan) [inline src/Spreadsheet.js]: Why did you delete this?

File: src/actions.js [hunk 3]
```diff
@@ -127,11 +127,15 @@ export const cut = (state: Types.StoreState<*>) => ({
   cut: true
 });
 
-export function paste<Cell: Types.CellBase>(state: Types.StoreState<Cell>) {
-  if (PointSet.isEmpty(state.copied)) {
-    return null;
-  }
-  const minPoint = PointSet.min(state.copied);
+export async function paste<Cell: Types.CellBase>(
```
Comment (iddan) [inline src/actions.js]: Actions should not do IO. Please return the readTextFromClipboard to the component.

File: src/Spreadsheet.js (hunk 2 above)
Comment (morloy) [reply src/Spreadsheet.js]: Since it always reads the the text from the clipboard, I figured this is not needed anymore.

File: src/actions.js (hunk 1 above)
Comment (morloy) [reply src/actions.js]: Using `Cell` causes a bunch of other errors. `any` was the best I could get, unfortunately.

**Comment by iddan** [comment]
Why did you remove the polyfill? A large portion of browsers does not support clipboard events yet.

**Comment by morloy** [comment]
> Why did you remove the polyfill? A large portion of browsers does not support clipboard events yet.

Not using events will results in the browser asking for permission to access the clipboard, which is quite a bad UX for my taste. The code in this PR is inspired from `nadbm/react-datasheet` and works in all modern browsers for me.
It seems to be quite well supported: https://caniuse.com/#feat=clipboard

**Comment by iddan** [comment]
Thank you very much for your contribution and patience getting this merged ❤️
